src/headerClassifier.ts

```diff
@@ -0,0 +1,152 @@
+        isLeaf: false,
+        level: i,
+        ptype:"None",
+        type: "String" // *pending* Incorporate user options *here*;
+    };
+    return Nnode;
+}
+
+function FormatErrors(rowname:string):void {
+    /*
+    * Detects unclosed parenthesis or parenthesis inside parenthesis
+    * Flags are used to signal the entrance to parenthesis
+        * { or [ => flag.On
+        * } or ] => flag.Off
+    * throws error if tries to raise an already raised flag or if it doesn"t lowers 
+      the flags at the end.
+    */
+   let Arrayflag:boolean = false;
+   let Objectflag:boolean = false;
+   for (var i:number = 0, tot:number = rowname.length; i < tot; i++) {
+        switch (rowname[i]) {
+            case "[":
+                if (Arrayflag || Objectflag){
+                    throw "Format Error, bad use of '['. Cause: "+rowname;
+                }
+                else {
+                    Arrayflag = true;
+                }
+                break;
+            case "{":
+                if (Arrayflag || Objectflag){
+                    throw "Format Error, bad use of '{'. Cause: "+rowname;
+                }
+                else {
+                    Objectflag = true;
+                }
+                break;
+            case "]":
+                if (Arrayflag && !Objectflag){
+                    Arrayflag = false;
+                }
+                else {
+                    throw "Format Error, bad use of ']'. Cause: "+rowname;
+                }
+                break;
+            case "}":
+                if (Objectflag && !Arrayflag){
+                    Objectflag = false;
+                }
+                else {
+                    throw "Format Error, bad use of '}'. Cause: "+rowname;
+                }
+                break;
+        }
+    }
+    if (Arrayflag || Objectflag) {
+        throw "Error, unclosed instance. Cause "+rowname;
+    }
+}
+
+function IndexErrors(nestMap:InestMap,rowname:string):void {
+    if (nestMap.tokens.length !== nestMap.modes.length) {
+        throw "Unexpected difference in the number of keys/indexes and modes. Cause: "+rowname;
+    }
+    for (var i:number = 0, tot:number = nestMap.tokens.length;i<tot;i++){
+        if (nestMap.modes[i] === "Array") {
+            if (isNaN(Number(nestMap.tokens[i]))) {
+                throw "Index Error, not a number. Cause: "+rowname;
+            }
+        }
+    }
+}
+
+export function rowClassify(rowname:string,Schem:object):Array<Inode>{
+    try {
+        FormatErrors(rowname);
+    } catch (error) {
+        console.log(error);
+    }
+    let rowHierarchy:Array<Inode> = [];
+    let parentList:Array<string|number> = [];
+    let nestMap:InestMap = tokenizeClassifier(rowname);
+
+    // check for number errors in nestMap for Array type
+    try {
+        IndexErrors(nestMap,rowname);
+    } catch (error) {
+        console.log(error);
+    }
+
+    for (var i:number = 0, tot:number = nestMap.modes.length;i<tot;i++){
```

Comment:
tot?

Reply:
tot = total, lo guardo para hacer el for a modo de no estar recalculándolo

Reply:
No usar `var` y debería llamarse 
modesLength
modesSize
modesCount
count
length
size

Reply:
cambiando